### PR TITLE
Do not copy __weakref__ from original _cls_dict

### LINKED ISSUE #407: Slots class holding a reference to the original version
I have a class hierarchy implemented with slots enabled, like this:

import attr

```python
@attr.s(slots=True)
class BaseClass(object):
    foo = attr.ib(default='foo')

@attr.s(slots=True)
class SubClass(BaseClass):
    bar = attr.ib(default='bar')
```

This works fine until I wanted to introspect the class hierarchy:

```python
BaseClass.__subclasses__()  # returns [<class '__main__.SubClass'>, <class '__main__.SubClass'>]
```

One of these is the original class, and the other is the new one created and returned by `attr.s`. This behavior manifests on both Python 3 and Python 2 (3.7.0 and 2.7.14, to be specific).

My understanding was that classes kept weak references to their subclasses to allow the `__subclasses__` method to work, and I know that attrs creates and returns a new class when `slots=True`, but it's not clear to me why the old class stays around. I don't see any obvious place where a strong reference to it is held.

I guessed that there might be a reference cycle somewhere, so I tried adding a call to `gc.collect()` and checking `gc.garbage`, but that turned out to be incorrect.

Because I didn't see that the reference-counting mentioned [in this comment](https://github.com/python-attrs/attrs/issues/102#issuecomment-252526692) was ever addressed, I also guessed that this might be a leak due to the `__class__` closure cell fixup. I decided that this isn't the cause, either, because Python 2 does not have the `__class__` closure cell.

### PR DESCRIPTION
**Pull request by gmacon**
This fixes #407 as discussed.

# Pull Request Check List

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](http://www.attrs.org/en/latest/contributing.html) at least once, it will save you unnecessary review cycles!

- [x] Added **tests** for changed code.
- [ ] New features have been added to our [Hypothesis testing strategy](https://github.com/python-attrs/attrs/blob/master/tests/strategies.py).
- [ ] Changes or additions to public APIs are reflected in our type stubs (files ending in ``.pyi``).
- [ ] Updated **documentation** for changed code.
- [x] Documentation in `.rst` files is written using [semantic newlines](http://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
- [x] Changes (and possible deprecations) have news fragments in [`changelog.d`](https://github.com/python-attrs/attrs/blob/master/changelog.d).

If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!

### CODE REVIEW

File: tests/test_make.py [hunk 1]
```diff
@@ -1250,6 +1251,25 @@ class C(object):
...
+        @attr.s(slots=True)
+        class C(object):
+            pass
+
+        @attr.s(slots=True)
+        class C2(C):
+            pass
+
+        # The original C2 is in a reference cycle, so force a collect:
+        gc.collect()
```
Comment (njsmith) [inline tests/test_make.py]: Sometimes calling `gc.collect()` once isn't enough to actually collect all garbage:
https://github.com/python-trio/trio/blob/45b4e586897f2676169b89e6b9f48d54eb14621a/trio/_core/tests/tutil.py#L26-L37
In this case I think it is probably fine but generally I feel like by the time you're calling `gc.collect` in a test you're already having a bad day and don't want to get surprised by any *new* edge cases.

Comment (gmacon) [reply tests/test_make.py]: That's interesting; I wasn't aware of that. IIUC, we'll be OK here because there are no `__del__` methods involved; does that sound right to you?

Comment (njsmith) [reply tests/test_make.py]: I was going to say "yes", but actually... How certain are we that there's no `__del__` involved in clearing the parent's subclasses table? (I suppose it probably uses weakrefs rather than `__del__`, but I'm not certain, and I'm not certain how weakrefs interact with `gc.collect` anyway.)
